File: infra-market-admin/src/api/activityTemplate.ts
import request from '@/utils/request'

// 活动模板相关类型定义
export interface ActivityTemplateField {
  name?: string
  label?: string
  type?: string
  inputType?: string
  required?: boolean
  defaultValue?: any
  description?: string
  sort?: number
  options?: SelectOption[]
  multiple?: boolean
  min?: number
  max?: number
  minLength?: number
  maxLength?: number
  pattern?: string
  placeholder?: string
  itemType?: ActivityTemplateField
  properties?: Record<string, ActivityTemplateField>
}

export interface SelectOption {
  value: string
  label?: string
}

export interface ActivityTemplate {
  id?: number
  name: string
  description?: string
  fields?: ActivityTemplateField[]
  status?: number
  createTime?: string
  updateTime?: string
}

export interface ActivityTemplateForm {
  id?: number
  name: string
  description?: string
  fields?: ActivityTemplateField[]
  status?: number
}

// 活动模板API
export const activityTemplateApi = {
  // 获取模板列表（支持分页）
  getList: (params?: {
    name?: string
    status?: number
    page?: number
    size?: number
  }) => {
    return request.get<PageResult<ActivityTemplate>>('/activity/template/list', { params })
  },

  // 获取所有启用的模板（用于下拉选择）
  getAll: () => {
    return request.get<ActivityTemplate[]>('/activity/template/all')
  },

  // 获取模板详情
  getById: (id: number) => {
    return request.get<ActivityTemplate>(`/activity/template/${id}`)
  },

  // 创建模板
  create: (data: ActivityTemplateForm) => {
    return request.post<ActivityTemplate>('/activity/template', data)
  },

  // 更新模板
  update: (id: number, data: ActivityTemplateForm) => {
    return request.put<ActivityTemplate>(`/activity/template/${id}`, data)
  },

  // 删除模板
  delete: (id: number) => {
    return request.delete<boolean>(`/activity/template/${id}`)
  },

  // 更新模板状态
  updateStatus: (id: number, status: number) => {
    return request.put<ActivityTemplate>(`/activity/template/${id}/status`, null, {
      params: { status }
    })
  },

  // 复制模板
  copy: (id: number) => {
    return request.post<ActivityTemplate>(`/activity/template/${id}/copy`)
  }
}

File: infra-market-server-vertx/src/main/kotlin/io/infra/market/vertx/entity/ActivityTemplate.kt
package io.infra.market.vertx.entity

/**
 * 活动模板实体类
 * 
 * 用于存储活动模板的基本信息和字段配置。
 * 活动模板定义了活动的配置结构，包括各种字段类型（整数、字符串、下拉框、数组等）。
 * 实现TimestampedEntity接口，提供创建时间和更新时间字段。
 * 
 * 表结构说明：
 * - 模板基本信息：名称、描述、状态等
 * - 字段配置：以JSON格式存储模板的所有字段配置
 * 
 * @author liuqinglin
 * @since 1.0.0
 */
data class ActivityTemplate(
    /**
     * 主键ID
     * 自增长主键，唯一标识一个活动模板记录
     */
    var id: Long? = null,
    
    /**
     * 模板名称
     * 模板的显示名称，用于界面展示和识别
     * 非空字段，建议长度不超过100字符
     */
    var name: String? = null,
    
    /**
     * 模板描述
     * 模板的详细说明，用于文档和用户理解
     * 可选字段，建议长度不超过1000字符
     */
    var description: String? = null,
    
    /**
     * 字段配置
     * 以JSON格式存储模板的所有字段配置
     * 包含字段类型、验证规则、默认值、选项等完整定义
     * 格式：[{"name":"field1","type":"string",...}, ...]
     */
    var fields: String? = null,
    
    /**
     * 模板状态
     * 1-启用，0-禁用
     * 用于控制模板的可用性，默认为1（启用）
     */
    var status: Int? = null,
    
    /**
     * 创建时间
     * 模板记录的创建时间（毫秒时间戳）
     */
    override var createTime: Long? = null,
    
    /**
     * 更新时间
     * 模板记录的最后修改时间（毫秒时间戳）
     */
    override var updateTime: Long? = null
) : TimestampedEntity
